AmplifyPlugins/Core/AWSPluginsCore/Model/Decorator/ModelIdDecorator.swift

```diff
@@ -18,7 +18,15 @@ public struct ModelIdDecorator: ModelBasedGraphQLDocumentDecorator {
         var fields = [String: String]()
         if let customPrimaryKeys = model.schema.customPrimaryIndexFields {
             for key in customPrimaryKeys {
-                fields[key] = model[key] as? String
+                if let date = model[key] as? Temporal.Date {
+                    fields[key] = date.iso8601String
+                } else if let dateTime = model[key] as? Temporal.DateTime {
+                    fields[key] = dateTime.iso8601String
+                } else if let time = model[key] as? Temporal.Time {
+                    fields[key] = time.iso8601String
+                } else {
+                    fields[key] = model[key] as? String
```

Comment:
the previous code assumed that the primary key can only be a string. 
1. are all values of the primary key just the ones you've listed here? (string, datetime, date, time). are there anything else? It doesn't make sense to fix it for one field type but not for others. 
2. based on 1. if it expands to more than (string, datetime, date, time), then is there already some code that allows you to extract the graphQL value out of the field? ie. can we reduce the code down to `fields[keys] = model[key].toGraphQLValue()`